plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt

```diff
@@ -625,7 +625,7 @@ class FeatureDevController(
                 return@withContext
             }
 
-            if (selectedFolder.parent.path != uri.path) {
+            if (!selectedFolder.path.startsWith(sourceFolder.path)) {
```

Comment:
Note: do not approve still, I noticed this is a naive approach, and a customer that changes a folder then cannot change the folder back to something else under the original folder. It is needed some context saving which was the original workspace root.